terra-jupyter-python/Dockerfile

```diff
@@ -35,6 +35,9 @@ RUN pip3 -V \
    pandas-gbq \
    seaborn \
    python-lzo \
+   # Hold back from the 3.0 upgrade, as it requires the BigQuery Storage API. See
+   # further in the Dockerfile for explanation of why BQ Storage is currently uninstalled.
+   "google-cloud-bigquery<3.0.0" \
```

Comment:
FYI @deflaux I'm layering this hack on top of the bigquery storage uninstall hack later in this Dockerfile. The ticket noted there is complete - however I'm pretty confident this permission was never backfilled for existing workspaces, and therefore it will cause issues in older workspaces if we remove the workaround. I don't have much context on the BigQuery magic issue we saw, so I'm not sure what other options exist here.

See https://github.com/googleapis/python-bigquery/blob/main/CHANGELOG.md#300-2022-03-29 on the BQ storage requirements in 3.0.0.